File: src/main/java/observer/sender/MailSender.java
package observer.sender;

import observer.Observer;

public class MailSender implements Observer {
    @Override
    public void update(String message) {
        sendMail(message);
    }

    private void sendMail(String message) {
        // 메일러 등의 서드파티 라이브러리로 메일 전송
        System.out.println("Mail : " + message);
    }
}

File: kotlin/src/main/kotlin/observer/sender/MailSender.kt
package observer.sender

import observer.Observer

class MailSender : Observer {
    override fun update(message: String) {
        sendMail(message)
    }

    private fun sendMail(message: String) {
        // 메일러 등의 서드파티 라이브러리로 메일 전송
        println("Mail : $message")
    }
}